front/lib/api/assistant/agent.ts

```diff
@@ -292,6 +295,26 @@ async function* runMultiActionsAgentLoop(
   }
 }
 
+async function getEmulatedAgentMessageActions(
+  auth: Authenticator,
+  {
+    agentMessage,
+    conversation,
+  }: { agentMessage: AgentMessageType; conversation: ConversationType }
+): Promise<AgentActionType[]> {
+  const actions: AgentActionType[] = [];
+  if (await isJITActionsEnabled(auth)) {
+    const a = makeConversationListFilesAction(agentMessage, conversation);
+    if (a) {
+      actions.push(a);
+    }
+  }
+
+  // We ensure that all emulated actions are injected with step -1.
+  assert(actions.every((a) => a.step === -1));
```

Comment:
nit: IMHO a regular exception with a clear error message is better (for eng runner) than using `assert` (which looks like something you'd use in a test)

Reply:
I guess I can add a message to the assert?